### DISCUSSION TITLE
host mode msc data transfer speed on rpi pico 2040

### DISCUSSION THREAD

**Discussion by mcl-uk** [hathach/tinyusb#2002] [Q&A]
Hello, I just got this demo code (https://github.com/rppicomidi/pico-usb-host-msc-demo) working on an RPi pico. I was disappointed by the low transfer speed (about 60kBytes/s) when reading large files from a usb memory stick. Does anyone know if there's any easy way to improve it? Should I expect similarly low speeds on other supported processors or are some much better than others? If so which processors should I be looking at?
Be grateful for any help, thanks for your time.

**Comment by hathach** [comment]
it is rp2040 specific driver issue at the moment https://github.com/hathach/tinyusb/issues/1766 . Currently I don't have time to implement the "fix". Other MCUs should work just fine.
